File: src/main/java/jp/co/sunarch/apps/idprovider/model/oauth2/password/ResourceOwnerPasswordCredentialsFlow.java
package jp.co.sunarch.apps.idprovider.model.oauth2.password;

import jp.co.sunarch.apps.idprovider.model.oauth2.EndpointRequestFactory;
import jp.co.sunarch.apps.idprovider.model.oauth2.TokenEndpoint;
import jp.co.sunarch.apps.idprovider.model.oauth2.AccessTokenResponse;

/**
 * @author takeshi
 */
public class ResourceOwnerPasswordCredentialsFlow implements TokenEndpoint<ResourceOwnerPasswordCredentialsAccessTokenRequest, AccessTokenResponse> {

  @Override
  public EndpointRequestFactory<ResourceOwnerPasswordCredentialsAccessTokenRequest> tokenRequestFactory() {
    // TODO
    return null;
  }

  @Override
  public AccessTokenResponse createToken(ResourceOwnerPasswordCredentialsAccessTokenRequest request) {
    // TODO
    return null;
  }

}

File: src/main/kotlin/jp/co/sunarch/apps/idprovider/model/oauth2/password/ResourceOwnerPasswordCredentialsFlow.kt
package jp.co.sunarch.apps.idprovider.model.oauth2.password

import jp.co.sunarch.apps.idprovider.model.oauth2.EndpointRequestFactory
import jp.co.sunarch.apps.idprovider.model.oauth2.TokenEndpoint
import jp.co.sunarch.apps.idprovider.model.oauth2.AccessTokenResponse

/**
 * @author takeshi
 */
class ResourceOwnerPasswordCredentialsFlow : TokenEndpoint<ResourceOwnerPasswordCredentialsAccessTokenRequest, AccessTokenResponse> {

  override fun tokenRequestFactory(): EndpointRequestFactory<ResourceOwnerPasswordCredentialsAccessTokenRequest> {
    TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
  }

  override fun createToken(request: ResourceOwnerPasswordCredentialsAccessTokenRequest): AccessTokenResponse {
    TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
  }
}
